src/extension.ts

```diff
@@ -104,45 +117,77 @@ export function activate(ctx: sourcegraph.ExtensionContext): void {
 
                 // TODO: This CSV generation is not robust.
                 const results = data.search.results.results
-                const csvData = [
-                    [
-                        'Repository',
-                        'Repository external URL',
-                        'File path',
-                        'File URL',
-                        'File external URL',
-                        'Search matches',
-                    ],
-                    ...results.map(r => {
-                        const searchMatches = r.lineMatches
-                            .map(line =>
-                                line.offsetAndLengths
-                                    .map(offset =>
-                                        line.preview?.substring(
-                                            offset[0],
-                                            offset[0] + offset[1]
-                                        )
+                const resultType = '__typename'
+                let csvData = new Array()
+
+                if (!results[0]) {
```

Comment:
Shouldn't we also check if `results` exist before accessing its item cause from the screenshot, it fails when trying map through `results` itself?  Also, if we want to check if there are results returned on not, maybe we could check through `.length`?

```js
if (!results?.length) {
    console.log('no results error')
}
```

Reply:
**Update**: I see that it works because previously, results could be null/undefined because the GraphQL query did not include other union types. Anyway, it would be nice to use the above condition if a new type/union is added to `SearchResult` in the future.

Reply:
If a new type/union is added to SearchResult in the future, I believe the default case in the switch statement would be able to catch it but I'm not sure, so I've added the `!results?.length` condition as well! Thank you so much @erzhtor !